helpers.js

```diff
@@ -42,9 +33,37 @@ function logToActivityLog(message) {
   console.error(message);
 }
 
+async function validatePaths(paths) {
+  const pathsArray = Array.isArray(paths) ? paths : [paths];
+
+  const pathPromises = pathsArray.map(pathExists);
+  const pathResults = await Promise.all(pathPromises);
```

Comment:
...then here we await `pathExistanceCheckPromises` and obtain `pathExistanceCheckResults`, and when we filter by that array in the line below, we know just by taking a single look at the filter function (without any additional contex), that it filters out non existing ones.